Main.cpp

```diff
@@ -191,43 +239,63 @@ int main() {
       // canBus.printRxAll();
     }
 
-    chThdSleepMilliseconds(50);
+    chThdSleepMilliseconds(10000);
   }
 }
 
-// module utility functions
 // @brief Convert bytes in RX buffer from ADC reading to temperature
 // @return temp Temperature in degrees C, resolution of .25
+// TODO: Add in per-channel moving average
 uint8_t adc_to_temp(uint8_t* rxbuf) {
   static uint16_t raw{};
-  static int norm{};
+  static double voltage{};
+  static int lookup_index{};
```

Comment:
A voltage below 1.48V will result in the expression the lookup index is assigned to to be negative. Obviously we can't use a negative index, and we don't need to support a voltage reading below 1.48V (the shutdown circuit will have tripped due to the fault) so the lower bound can just be clamped to zero. I could however move this to the voltage var itself, which would make more sense.